### PR TITLE
Removed usage of stream.available for stream size (it is the entire size in some implementations but it is not guaranteed)

### PR DESCRIPTION
**Pull request by mrydzy**
```
* Checking file size for JarUrlConnection and FileUrlConnection
* Sending chuncked response otherwise
```

### CODE REVIEW

File: framework/src/play/src/main/scala/play/api/controllers/Assets.scala [hunk 1]
```diff
@@ -475,6 +489,18 @@ class AssetsBuilder(errorHandler: HttpErrorHandler) extends Controller {
     }
   }
 
+  def getResult(file: String, assetInfo: AssetInfo, connection: URLConnection, gzipRequested: Boolean): Result = {
```
Comment (jroper) [inline framework/src/play/src/main/scala/play/api/controllers/Assets.scala]: Make this private, also perhaps rename to `createResult`.

File: framework/src/play/src/main/scala/play/api/controllers/Assets.scala [hunk 2]
```diff
@@ -379,7 +381,7 @@ class AssetsBuilder(errorHandler: HttpErrorHandler) extends Controller {
   }
 
   private def result(file: String,
```
Comment (jroper) [inline framework/src/play/src/main/scala/play/api/controllers/Assets.scala]: Perhaps rename to `resultWithContentLength`.

File: framework/src/play/src/main/scala/play/api/controllers/Assets.scala [hunk 3]
```diff
@@ -394,6 +396,22 @@ class AssetsBuilder(errorHandler: HttpErrorHandler) extends Controller {
         )
       ),
       resourceData)
+    resultWithCompressionHeaders(gzipRequested, gzipAvailable, response)
+  }
+
+  private def result(file: String,
```
Comment (jroper) [inline framework/src/play/src/main/scala/play/api/controllers/Assets.scala]: Perhaps rename to `chunkedResult`.

**Comment by jroper** [comment]
We should probably have some tests that assert the content length is right - they can be added to `framework/src/play-integration-test/src/test/scala/play/it/http/assets/AssetsSpec.scala`.  I don't think any of the existing tests check for a content length header.

Also, would be good to test both the file case and the jar case.  All the existing tests there, in practice, are testing the file case.  You'll need to come up with a resource that can be served from a jar off the classpath. This may best be achieved by creating a custom `URLClassLoader` that gets passed to the `ServerConfig` constructor used in that spec, that zips up the `testresources` directory to ensure it's on the classpath.  Let me know if you need help doing that.

**Comment by mrydzy** [comment]
I'm not sure if I understood why to create custom URLClassLoader. 
Maybe I could simply use JarOutputStream to create some jar file?

**Comment by benmccann** [comment]
@mrydzy are you still interested in getting this change in? if so, it would be helpful to rebase the PR and add some tests

**Comment by mkurz** [comment]
Ping @mrydzy

**Comment by wsargent** [comment]
closing due to lack of activity

**Pull request closed without merging**